components/autofill/src/db/store.rs

```diff
@@ -105,6 +105,16 @@ impl Store {
         self.store_impl.touch_address(guid)
     }
 
+    pub fn scrub_encrypted_data(&self) -> Result<()> {
+        // scrub the data on disk
+        self.store_impl.scrub_encrypted_data()?;
```

Comment:
This function should just call the impl, so please move the code below down into that impl.

Reply:
The impl definitely seems like a better place, but `create_engine()` inputs `Arc<StoreImpl>` not `&StoreImpl` , and I couldn't figure out how to do that in the StoreImpl method. @lougeniaC64 maybe we could discuss this one some more today.

Reply:
@mhammond Were you thinking something like this in the `scrub_encrypted_data` impl function?

```
crate::sync::credit_card::create_engine(Arc::new(*self)).reset_local_sync_data()?;
// also tried self.clone() with StoreImpl implementing Clone
```

This won't compile because the `writer` field of `AutofilDb` doesn't implement Clone.

Reply:
This is maybe out of scope for this issue, but it feels like the right way to do this would be to create a `SyncEngineStorageImpl` directly and call a method on that.  However, that's not currently possible because `put_meta()` is implemented in `ConfigSyncEngine`.  Would it make sense to move the put/get/delete meta functionality into `SyncEngineStorageImpl`?  I guess we would still need to wrap the methods in `ConfigSyncEngine` but that wouldn't be too bad.

Reply:
right, I missed that detail. Just leaving it where it is is fine for now.  (It does seem a little heavy-handed that we need to create an engine just for that reset, but it's not a huge deal and isn't a reliability or performance concern so can be cleaned up later)